packages/core/test/userapp/index.js

```diff
@@ -507,20 +507,18 @@ const addRequestHeader = (request, z, bundle) => {
  */
 const changeStatusOnErrorResponses = response => {
   const contentType = response.getHeader('Content-Type');
-  if (!contentType) {
-    return response;
-  }
 
-  const isJsonResponse = contentType.match(/^application\/json/);
-  if (!isJsonResponse) {
-    return response;
+  if (contentType && contentType.match(/^application\/json/)) {
+    const data = JSON.parse(response.content);
+    const error = data.args.error;
+    if (response.status === 200 && error) {
+      response.status = 500;
+    }
   }
 
-  const data = JSON.parse(response.content);
-  const error = data.args.error;
-  if (response.status === 200 && error) {
-    response.status = 500;
-  }
+  // defining `afterResponse` makes use responsible for calling `throwForStatus()`
+  response.throwForStatus();
```

Comment:
Our test app was a good example of an app that now needs to to take its responsibility for error handling.